pkg/controllers/status/work_status_controller.go

```diff
@@ -294,6 +294,12 @@ func (c *WorkStatusController) handleDeleteEvent(ctx context.Context, key keys.F
 		return nil
 	}
 
+	// skip processing as the work object is suspended for dispatching.
+	if helper.IsWorkSuspendDispatching(work) {
+		return nil
+	}
+
+	//nolint:staticcheck // SA1019 ignore deprecated util.PropagationInstruction
```

Comment:
The `PropagationInstruction` may not have been deprecated in version 1.13, so this line can be removed.